pkgs/misc/vim-plugins/update.py

```diff
@@ -386,6 +416,31 @@ def generate_nix(plugins: List[Tuple[str, str, Plugin]], outfile: str):
     print(f"updated {outfile}")
 
 
+def update_redirects(input_file: Path, output_file: Path, redirects: dict):
+    with fileinput.input(input_file, inplace=True) as f:
+        for line in f:
+            print(redirects.get(line, line), end="")
+    print(
+        f"""\
+Redirects have been detected and {input_file} has been updated. Please take the
+following steps:
+    1. Go ahead and commit just the updated expressions as you intended to do:
+            git add {output_file}
+            git commit -m "vimPlugins: Update"
+    2. If any of the plugin names were changed, add the old names as aliases in
+    aliases.nix
```

Comment:
It would be amazing if we could automate this as well, though that's not a requirement for merge in my view (but a very nice to have).

Reply:
I'd rather invest effort into eliminating aliases.nix if possible.

Reply:
I would propose to keep that file (or an equivalent file) but just add entries in the form of

```
old_name = throw "${old_name} was renamed to ${new_name}. Please use that instead" # added on yyyy-dd-mm
```

Since just doing the rename without any notification would be confusing to end users. We could then drop the throw lines once they're "old enough" (maybe 2 releases or something, see https://github.com/NixOS/rfcs/pull/33).

So either way we'd still need a file that keeps track of name changes.

Reply:
I added a commit to throw errors like this rather than silently aliasing.